source/MaterialXTest/ShaderValid.cpp

```diff
@@ -883,13 +909,22 @@ TEST_CASE("MaterialX documents", "[shadervalid]")
     // This will be imported in each test document below
     ioTimer.startTimer();
     mx::DocumentPtr dependLib = mx::createDocument();
-    loadLibraries({ "stdlib", "sxpbrlib" }, searchPath, dependLib);
-#ifdef MATERIALX_BUILD_GEN_GLSL
-    if (options.runGLSLTests)
+    std::set<std::string> excludeFiles;
+    if (!options.runGLSLTests)
     {
-        glslColorManagementSystem->loadLibrary(dependLib);
+        excludeFiles.insert("stdlib_sx-glsl_impl.mtlx");
     }
-#endif
+    if (!options.runOSLTests)
+    {
+        excludeFiles.insert("stdlib_osl_impl.mtlx");
+        excludeFiles.insert("stdlib_sx-osl_impl.mtlx");
+    }
+    if (options.cmsFiles.size() == 0)
+    {
+        excludeFiles.insert("cm_impl.mtlx");
+    }
+    const mx::StringVec libraries = { "stdlib", "sxpbrlib" };
+    loadLibraries(libraries, searchPath, dependLib, &excludeFiles);
```

Comment:
loadLibraries has the new optional excludeFiles arg to filter out loading in undesired files.